sdk-commons/src/main/java/org/openmrs/maven/plugins/utility/DistroHelper.java

```diff
@@ -299,213 +303,152 @@ public void saveDistroPropertiesTo(File destination, String distro) throws MojoE
-	 * against the {@code DistroProperties} provided. This method ensures that the dependencies defined
-	 * in the {@code content.properties} file are either present in the {@code distroProperties} file with
-	 * a version that matches the specified version range, or it finds the latest matching version if not
-	 * already specified in {@code distroProperties}.
-	 *
-	 * <p>The method iterates over each dependency listed in the {@code content.properties} file, focusing on
-	 * dependencies that start with specific prefixes such as {@code omod.}, {@code owa.}, {@code war},
-	 * {@code spa.frontendModule}, or {@code content.}. For each dependency, the method performs the following:</p>
-	 * <ul>
-	 *   <li>If the dependency is not present in {@code distroProperties}, it attempts to find the latest version
-	 *   matching the specified version range and adds it to {@code distroProperties}.</li>
-	 *   <li>If the dependency is present, it checks whether the version specified in {@code distroProperties}
-	 *   falls within the version range specified in {@code content.properties}. If it does not, an error is thrown.</li>
-	 * </ul>
-	 *
-	 * @param contentProperties The {@code Properties} object representing the {@code content.properties} file
-	 *                          of a content package.
-	 * @param distroProperties  The {@code DistroProperties} object containing key-value pairs specifying
-	 *                          the current distribution's dependencies and their versions.
-	 * @param zipFileName       The name of the ZIP file containing the {@code content.properties} file being processed.
-	 *                          Used in error messages to provide context.
-	 *
-	 * @throws MojoExecutionException If no matching version is found for a dependency not defined in
-	 *                                {@code distroProperties}, or if the version specified in {@code distroProperties}
-	 *                                does not match the version range in {@code content.properties}.
+	 * @return the missing dependencies for a given frontend artifact.
 	 */
-	protected void processContentProperties(Properties contentProperties, DistroProperties distroProperties, String zipFileName) throws MojoExecutionException {
-		for (String dependency : contentProperties.stringPropertyNames()) {
-			if (dependency.startsWith("omod.") || dependency.startsWith("owa.") || dependency.startsWith("war")
-					|| dependency.startsWith("spa.frontendModule") || dependency.startsWith("content.")) {
-				String versionRange = contentProperties.getProperty(dependency);
-				String distroVersion = distroProperties.getParam(dependency);
-
-				if (distroVersion == null) {
-					String latestVersion = findLatestMatchingVersion(dependency, versionRange);
-					if (latestVersion == null) {
-						throw new MojoExecutionException(
-								"No matching version found for dependency " + dependency + " in " + zipFileName);
```

Comment:
This method is a bit of a hack to support what PIH is currently doing, which is building our frontend in a separate project and publishing it to Maven, and then including that Maven zip in our distribution.  In this case, we don't have the spa-assemble or spa-build config files in that artifact (though as an alternative to this below, we could potentially add it in).  This downloads the artifact, unpacks it, and inspects the file/directory contents, translating the directory names back into ESM names and versions.  Interested in review/thoughts on this in particular @ibacher .  There is an integration test in `DistroHelperIT` that demonstrates this.